mlflow/projects/databricks.py

```diff
@@ -120,20 +120,20 @@ def _upload_project_to_dbfs(self, project_dir, experiment_id):
                             a directory containing an MLproject file).
         """
         temp_tarfile_dir = tempfile.mkdtemp()
-        temp_tar_filename = os.path.join(temp_tarfile_dir, "project.tar.gz")
+        temp_tar_filename = posixpath.join(temp_tarfile_dir, "project.tar.gz")
```

Comment:
```suggestion
        temp_tar_filename = os.path.join(temp_tarfile_dir, "project.tar.gz")
```

A little tricky, but this path is on the local filesystem so we should use the OS's path join logic here